activesupport/lib/active_support/testing/assertions.rb

```diff
@@ -181,12 +181,24 @@ def assert_no_difference(expression, message = nil, &block)
       #
       # The keyword arguments +:from+ and +:to+ can be given to specify the
       # expected initial value and the expected value after the block was
-      # executed.
+      # executed. The comparison is done using case equality (===), which means
+      # you can specify patterns or classes:
       #
+      #   # Exact value match
       #   assert_changes :@object, from: nil, to: :foo do
       #     @object = :foo
       #   end
       #
+      #   # Pattern match
```

Comment:
I think it's better to change the wording here to avoid confusion with [Ruby pattern matching](https://docs.ruby-lang.org/en/3.4/syntax/pattern_matching_rdoc.html)
```suggestion
      #   # Case equality
```

Reply:
Thank you for the feedback! You're absolutely right - we should avoid conflating the `===` semantics with Ruby's formal pattern-matching syntax.